File: src/main/java/org/nitk/hackathon/utils/NetBankingPayment.java
package org.nitk.hackathon.utils;

public class NetBankingPayment implements Payment {
    private String gatewayName;
    private String accountNumber;
    private PaymentGateway paymentGateway;

    public NetBankingPayment(String gatewayName, String accountNumber) {
        this.gatewayName = gatewayName;
        this.accountNumber = accountNumber;        
    }

    @Override
    public void pay(double amount) {
        if(null == paymentGateway) {
            paymentGateway = new PaymentGateway(gatewayName, accountNumber);
        }
        paymentGateway.pay(amount);
        System.out.println("Paid " + amount + " via Net Banking");
    }

    @Override
    public Object copy() {
        return null;
    }
}

File: python/utils/netbanking_payment.py

from python.utils.payment import Payment
from python.utils.payment_gateway import PaymentGateway


class NetBankingPayment(Payment):
    def __init__(self, gateway_name, account_number):
        self.gateway_name = gateway_name
        self.account_number = account_number
        self.payment_gateway = None

    def pay(self, amount):
        if self.payment_gateway is None:
            self.payment_gateway = PaymentGateway(self.gateway_name, self.account_number)
        self.payment_gateway.pay(amount)
        print(f"Paid {amount} via Net Banking")

    def copy(self):
        return self.copy(self) 
